distributed/ddp/example.py

```diff
@@ -53,11 +54,30 @@ def spmd_main(local_world_size, local_rank):
         key: os.environ[key]
         for key in ("MASTER_ADDR", "MASTER_PORT", "RANK", "WORLD_SIZE")
     }
-    print(f"[{os.getpid()}] Initializing process group with: {env_dict}")
-    dist.init_process_group(backend="nccl")
+    
+    if sys.platform == "win32":
+        # Distributed package only covers collective communications with Gloo
+        # backend and FileStore on Windows platform. Set init_method parameter
+        # in init_process_group to a local file.
+        if "INIT_METHOD" in os.environ.keys():
+            print(f"init_method is {os.environ['INIT_METHOD']}")
+            url_obj = urlparse(os.environ["INIT_METHOD"])
+            if url_obj.scheme.lower() != "file":
+                raise ValueError("Windows only supports FileStore")
+            else:
+                init_method = os.environ["INIT_METHOD"]
+        else:
+            # It is a example application, For convience, we create a file in temp dir.
+            temp_dir = tempfile.gettempdir()
+            init_method = "file:///" + os.path.join(temp_dir, "ddp_example")
+        dist.init_process_group(backend="gloo", init_method=init_method, rank=int(env_dict["RANK"]), world_size=int(env_dict["WORLD_SIZE"]))
+    else:
+        print(f"[{os.getpid()}] Initializing process group with: {env_dict}")  
+        dist.init_process_group(backend="nccl")
+
     print(
         f"[{os.getpid()}]: world_size = {dist.get_world_size()}, "
-        + f"rank = {dist.get_rank()}, backend={dist.get_backend()}"
+        + f"rank = {dist.get_rank()}, backend={dist.get_backend()} \n"
```

Comment:
Can you please explain, why this is needed? (print already adds a new line at the end of the statement, isn't it?) 
```suggestion
        + f"rank = {dist.get_rank()}, backend={dist.get_backend()}"
```

Reply:
Thank your question. In practice, both in Window and Linux, I often find the newlines are skipped in the distributed examples. There's an answer in [stackoverflow](https://stackoverflow.com/questions/31142566/print-skipping-newline). 

But it would generate some redundant newlines.
So I made a little change to fully control the newlines in latest commit. 
example: print("xxxx \n", end='')